### ISSUE TITLE
The link modal posts list suggestion is not accessible

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#2083]
The link modal posts search / suggestions merged via #1985 

<img width="314" alt="screen shot 2017-07-24 at 13 58 25" src="https://user-images.githubusercontent.com/272444/28524397-38a61350-7078-11e7-85ee-0e19fab74f5b.png">

should be implemented as an ARIA combobox widget + listbox as done for the Tags suggestion. This was originally noted in the related issue #1040 but the implementation missed to address any accessibility feature.

The PR mentions this:
> This needs some Accessibility eyes, We can navigate using arrows but I guess this is far from perfect.

So I'm opening a specific issue. The UI should basically replicate what was already done for the Tag suggestions, see #1141. I hope the great work done there can be re-used easily. Worth also considering some abstraction.

Worth noting the current implementation is actually an a11y regression compared to the current functionality in WordPress.
